eclair-core/src/test/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreClientSpec.scala

```diff
@@ -1374,39 +1491,195 @@ class BitcoinCoreClientSpec extends TestKitBaseClass with BitcoindService with A
+      assert(sender.expectMsgType[Failure].cause.getMessage.contains("Transaction not in mempool"))
+    })
+
+    // The wallet transactions have been evicted, but they must be kept inside the wallet and not be double-spent.
+    val txNotFunded = Transaction(2, Nil, Seq(TxOut(50_000 sat, Script.pay2wpkh(randomKey().publicKey))), 0)
+    wallet.fundTransaction(txNotFunded, FeeratePerKw(5000 sat), replaceable = true).pipeTo(sender.ref)
     assert(sender.expectMsgType[Failure].cause.getMessage.contains("Insufficient funds"))
-    // The transaction is kept in bitcoind's internal wallet.
-    wallet.rpcClient.invoke("gettransaction", tx.txid).map(json => Transaction.read((json \ "hex").extract[String])).pipeTo(sender.ref)
-    assert(sender.expectMsgType[Transaction].txid == tx.txid)
+    Seq(spliceTx, anchorTx).foreach(walletTx => {
+      wallet.rpcClient.invoke("gettransaction", walletTx.txid).map(json => Transaction.read((json \ "hex").extract[String])).pipeTo(sender.ref)
+      assert(sender.expectMsgType[Transaction].txid == walletTx.txid)
+    })
+
+    // We now double-spend parent transactions, which permanently invalidates our wallet transactions:
+    //                   +------------------+     +------------------+     +-----------+
+    // funding_tx_1 +--->| remote_splice_tx |---->| remote_splice_tx |---->|           |
+    //              |    +------------------+     +------------------+     | splice_tx |
+    //              |    +--------------------+         wallet_utxo_1 ---->|           |
+    //              +--->| remote_commit_tx_1 |                            +-----------+
+    //                   +--------------------+
+    //
+    //                   +-----------------+     +-----------+
+    // funding_tx_2 +--->| local_commit_tx |---->|           |
+    //              |    +-----------------+     | anchor_tx |
+    //              |         wallet_utxo_2 ---->|           |
+    //              |                            +-----------+
+    //              |    +--------------------+
+    //              +--->| remote_commit_tx_2 |
+    //                   +--------------------+
+    val remoteCommitTx1 = spendFundingScript(fundingUtxos(0), 500_000 sat, 400_000 sat)
+    val remoteCommitTx2 = spendFundingScript(fundingUtxos(1), 500_000 sat, 400_000 sat)
+    generateBlocks(1)
+
+    // Both funding outputs have been spent by transactions that are external to our wallet.
+    fundingUtxos.foreach(utxo => {
+      wallet.isTransactionOutputSpendable(utxo.txid, utxo.index.toInt, includeMempool = false).pipeTo(sender.ref)
+      sender.expectMsg(false)
+    })
+    Seq(remoteCommitTx1, remoteCommitTx2).foreach(tx => {
+      wallet.getTxConfirmations(tx.txid).pipeTo(sender.ref)
+      sender.expectMsg(Some(1))
+    })
+
+    // Wallet transaction are still kept inside the wallet and cannot be double-spent, even though they have been
+    // permanently double-spent.
```

Comment:
```suggestion
    // permanently double-spent at their ancestors' level.
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2817/commits/5ffbdb67e576ba54b3ee43be1217b9c1fba57a54